### PR TITLE
CMR-4227

### PR DESCRIPTION
**Pull request by laurenfrederick**
Wrote a test framework to be able to automatically run through a range of boost values for a given field. Had to do some refactoring of the test framework for this so you'll see some stuff moved around.

Changed the entry-title boost value.

### CODE REVIEW

File: search-relevancy-test/src/search_relevancy_test/result_logger.clj [hunk 1]
```diff
@@ -18,12 +18,17 @@
...
   "Read the CSV file and get the data from the last run, if exists"
   [csv-file]
-  (when-let [last-run (->> (core/read-csv local-test-run-csv)
+  (when-let [last-run (->> (core/read-csv csv-file)
                            (map #(assoc % :date (time-format/parse (:date %))))
                            (group-by :date)
                            sort
                            last)]
-    (val last-run)))
+    (->> last-run
```
Comment (oubiwann) [inline search-relevancy-test/src/search_relevancy_test/result_logger.clj]: Nice use of threading macro.

File: search-relevancy-test/src/search_relevancy_test/result_logger.clj [hunk 2]
```diff
@@ -96,20 +101,16 @@
...
-      (write-test-run-to-csv test-run-history-csv results run-description)
-      (write-test-run-to-csv local-test-run-csv results run-description))
-    (write-test-run-to-csv local-test-run-csv results nil)))
+  (let [run-description (core/get-argument-value args "-log-run-description")
+        log-history (core/get-argument-value args "-log-history")
+        _ (def args args)
+        log-history (if (some? log-history)
+                      (Boolean/parseBoolean log-history)
+                      true)]
+    (if log-history
```
Comment (oubiwann) [inline search-relevancy-test/src/search_relevancy_test/result_logger.clj]: With one condition, you should use `when` instead.

File: search-relevancy-test/src/search_relevancy_test/runner.clj [hunk 3]
```diff
@@ -3,91 +3,30 @@
...
-   [clj-http.client :as client]
-   [clojure.java.io :as io]
    [clojure.string :as string]
    [search-relevancy-test.anomaly-fetcher :as anomaly-fetcher]
-   [search-relevancy-test.core :as core]
-   [search-relevancy-test.ingest :as ingest]
-   [search-relevancy-test.reporter :as reporter]
-   [search-relevancy-test.result-logger :as result-logger]
-   [clojure.set :as set]))
+   [search-relevancy-test.boost-test :as boost-test]
```
Comment (oubiwann) [inline search-relevancy-test/src/search_relevancy_test/runner.clj]: Good organization.

File: search-relevancy-test/src/search_relevancy_test/boost_test.clj [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+   "-log-history"
+   "false"])
+
+(defn- run-boost-test
+  "Run the relevancy tests with the boost for the given field configured through
+  search params. Return the result summary with the boost value assoc'd so we can
+  reference it later"
+  [field boost]
+  (let [params (create-search-params field boost)] ; append to search request
+    (println "***********************************************")
```
Comment (oubiwann) [inline search-relevancy-test/src/search_relevancy_test/boost_test.clj]: You're using this string separator in other places -- I'd suggest a `def` at the top of the file (or, if used in multiple namespaces, with other common constants).

File: search-relevancy-test/src/search_relevancy_test/result_logger.clj (hunk 2 above)
Comment (laurenfrederick) [inline search-relevancy-test/src/search_relevancy_test/result_logger.clj]: Good catch, I changed the logic but forgot to change that. Done.

File: search-relevancy-test/src/search_relevancy_test/boost_test.clj (hunk 4 above)
Comment (laurenfrederick) [inline search-relevancy-test/src/search_relevancy_test/boost_test.clj]: Yeah I was kind of haphazardly separating things. I changed it.

**Pull request merged**